Sources/TuistDependencies/Carthage/Interactors/CarthageFrameworksInteractor.swift

```diff
@@ -0,0 +1,81 @@
+import Foundation
+import TSCBasic
+import TuistCore
+import TuistSupport
+
+/// Saving frameworks installed using `carthage`.
+public protocol CarthageFrameworksInteracting {
+    /// Saves frameworks installed using `carthage`.
+    /// - Parameters:
+    ///   - carthageBuildDirectory: The path to the directory that contains frameworks built by Carthage.
+    ///   - destinationDirectory: The path to the directory where frameworks built by Carthage should be saved.
+    func copyFrameworks(carthageBuildDirectory: AbsolutePath, destinationDirectory: AbsolutePath) throws
+}
+
+public final class CarthageFrameworksInteractor: CarthageFrameworksInteracting {
+    private let fileHandler: FileHandling
+
+    public init(fileHandler: FileHandling = FileHandler.shared) {
+        self.fileHandler = fileHandler
+    }
+
+    public func copyFrameworks(carthageBuildDirectory: AbsolutePath, destinationDirectory: AbsolutePath) throws {
+        let alreadyInstalledFrameworks: Set<String> = try getAlreadyInstalledFrameworkNames(at: destinationDirectory)
+
+        try Platform.allCases.forEach { platform in
```

Comment:
I'd recommend getting rid of this class altogether and update the `CarthageInteractor` to delete all the Carthage frameworks from the destination directory and move the new ones that have been built.  We can improve the logic in the future using the Cartfile.resolved to do some diffing.